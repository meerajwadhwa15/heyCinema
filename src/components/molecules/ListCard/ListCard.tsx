import React, { FunctionComponent } from 'react';
import Img from './../../atoms/Img';
import Anchor from './../../atoms/Anchor';

export type Props = {
  item: {
    imdbID: string;
    Title: string;
    Year: string;
    Type: string;
    Poster: string;
  };
};

const BeerList: FunctionComponent<Props> = ({ item }) => (
  <article className="card">
    <Anchor url={`/${item.imdbID}`} title={item.Title}>
      <div>
        <Img src={item.Poster} alt={item.Title} />
      </div>
      <div>
        <h3>{item.Title}</h3>
        <p>{item.Type}</p>
        <p>{item.Year}</p>
      </div>
    </Anchor>
  </article>
);

export default BeerList;
