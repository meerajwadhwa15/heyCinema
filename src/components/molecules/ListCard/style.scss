.card {
  background: $secondaryBackgroundColor;
  margin-bottom: 15px;
  border: 1px solid $tertiaryColor;
  border-radius: 10px;

  &:hover {
    background: $hoverEffectColor;
  }

  h3 {
    color: $primaryColor;
  }

  a {
    display: flex;
    flex-direction: column;
    text-decoration: none;
    color: $secondaryColor;

    > div:last-child {
      padding: 15px;
      border-top: 1px solid $primaryBackgroundColor;
    }

    > div:first-child {
      border-radius: 10px 0 0 10px;
      text-align: center;
    }

    img {
      max-height: 400px;
      width: auto;
      height: auto;
    }

    @media only screen and (min-width: $min-sm-screen) {
      flex-direction: row;

      > div:first-child {
        overflow: hidden;
      }

      > div:last-child {
        border-top: none;
      }

      img {
        width: 180px;
        height: 250px;

        &:hover {
          transition: transform 0.5s ease;
          transform: scale(1.1);
        }
      }
    }
  }

  p {
    margin-top: 15px;
  }
}
