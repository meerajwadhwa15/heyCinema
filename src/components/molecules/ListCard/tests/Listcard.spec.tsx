import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ListCard, { Props } from '../ListCard';

const props: Props = {
  item: {
    imdbID: 'id',
    Title: 'title',
    Year: '2019',
    Type: 'test',
    Poster: 'test'
  }
};

describe('ListCard Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(
      <BrowserRouter>
        <ListCard {...props} />
      </BrowserRouter>
    );
  });
});

describe('ListCard Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(
      <BrowserRouter>
        <ListCard {...props} />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
