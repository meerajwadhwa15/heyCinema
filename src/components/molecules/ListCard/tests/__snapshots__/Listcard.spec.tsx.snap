// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard Snapshot testing Matches the snapshot 1`] = `
<article
  className="card"
>
  <a
    href="/id"
    onClick={[Function]}
    title="title"
  >
    <div>
      <img
        alt="title"
        onError={[Function]}
        src="test"
      />
    </div>
    <div>
      <h3>
        title
      </h3>
      <p>
        test
      </p>
      <p>
        2019
      </p>
    </div>
  </a>
</article>
`;
